lib/win32/security/sid.rb

```diff
@@ -255,13 +255,13 @@ def initialize(account=nil, host=Socket.gethostname)
           end
         else
           bool = LookupAccountName(
-            host,
-            account,
-            sid,
-            sid_size,
-            domain,
-            domain_size,
-            use_ptr
+              host.encode("UTF-16LE"),
```

Comment:
We use special helpers to convert strings due to Ruby bugs - see https://github.com/puppetlabs/puppet/blob/master/lib/puppet/util/windows/string.rb